packages/features/troubleshooter/components/EventTypeSelect.tsx

```diff
@@ -0,0 +1,53 @@
+import { useMemo, useEffect } from "react";
+
+import { trpc } from "@calcom/trpc";
+import { SelectField } from "@calcom/ui";
+
+import { useTroubleshooterStore } from "../store";
+
+export function EventTypeSelect() {
+  const { data: eventTypes, isLoading } = trpc.viewer.eventTypes.list.useQuery();
+  const selectedEventType = useTroubleshooterStore((state) => state.event);
+  const setSelectedEventType = useTroubleshooterStore((state) => state.setEvent);
+
+  // const selectedEventQueryParam = getQueryParam("eventType");
+
+  const options = useMemo(() => {
+    if (!eventTypes) return [];
+    return eventTypes.map((e) => ({
+      label: e.title,
+      value: e.slug,
+      id: e.id,
+      duration: e.length,
+    }));
+  }, [eventTypes]);
+
+  useEffect(() => {
+    if (!selectedEventType && eventTypes && eventTypes[0]) {
+      const { id, slug, length } = eventTypes[0];
+      setSelectedEventType({
+        id,
+        slug,
+        duration: length,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [eventTypes]);
```

Comment:
This looks weird - but is needed to automatically set the eventType if there is no exisitng query param